sumneko 3.1.0 breaks under darwin

### Describe the bug
Building the most recent version breaks under darwin

### Steps To Reproduce
`nix-env -ia nixpkgs.sumneko-lua-language-server`

### Additional context
Full log:
```
building '/nix/store/lzp8567dcngr7g69h8468d858aihs15j-sumneko-lua-language-server-3.1.0.drv'...
unpacking sources
unpacking source archive /nix/store/7z0zgvqps00zz5n1s8qxlhvcf1ygkf5s-source
source root is source
patching sources
configuring
no configure script, doing nothing
building
build flags: -j12 -l12 -fcompile/ninja/macos.ninja
[1/60] Compile C   build/macos/obj/source_bee/lua-seri.obj
[2/60] Compile C++ build/macos/obj/source_bee/error.obj
[3/60] Compile C++ build/macos/obj/source_bee/fsevent_osx.obj
[4/60] Compile C++ build/macos/obj/source_bee/endpoint.obj
[5/60] Compile C++ build/macos/obj/source_bee/socket.obj
[6/60] Compile C++ build/macos/obj/source_bee/format.obj
[7/60] Compile C++ build/macos/obj/source_bee/os.obj
[8/60] Compile C++ build/macos/obj/source_bee/subprocess_posix.obj
FAILED: build/macos/obj/source_bee/subprocess_posix.obj
clang++ -MMD -MT build/macos/obj/source_bee/subprocess_posix.obj -MF build/macos/obj/source_bee/subprocess_posix.obj.d -std=c++17 -O2 -Wall -Werror -fvisibility=hidden -mmacosx-version-min=10.15 -I3rd/bee.lua/bee/nonstd -I3rd/bee.lua -DBEE_INLINE -DNDEBUG -Wunguarded-availability -o build/macos/obj/source_bee/subprocess_posix.obj -c 3rd/bee.lua/bee/subprocess/subprocess_posix.cpp
3rd/bee.lua/bee/subprocess/subprocess_posix.cpp:200:13: error: use of undeclared identifier 'posix_spawn_file_actions_addchdir_np'; did you mean 'posix_spawn_file_actions_addinherit_np'?
            posix_spawn_file_actions_addchdir_np(&spawnfile_, cwd);
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            posix_spawn_file_actions_addinherit_np
/nix/store/x9qw47hy8mpczwbvb03vyyn8qb2v27n7-Libsystem-1238.60.2/include/spawn.h:143:5: note: 'posix_spawn_file_actions_addinherit_np' declared here
int     posix_spawn_file_actions_addinherit_np(posix_spawn_file_actions_t *,
        ^
3rd/bee.lua/bee/subprocess/subprocess_posix.cpp:200:63: error: cannot initialize a parameter of type 'int' with an lvalue of type 'const char *'
            posix_spawn_file_actions_addchdir_np(&spawnfile_, cwd);
                                                              ^~~
/nix/store/x9qw47hy8mpczwbvb03vyyn8qb2v27n7-Libsystem-1238.60.2/include/spawn.h:144:6: note: passing argument to parameter here
                int) __OSX_AVAILABLE_STARTING(__MAC_10_7, __IPHONE_4_3);
                   ^
2 errors generated.
ninja: build stopped: subcommand failed.
error: builder for '/nix/store/lzp8567dcngr7g69h8468d858aihs15j-sumneko-lua-language-server-3.1.0.drv' failed with exit code 1;
       last 10 log lines:
       > int     posix_spawn_file_actions_addinherit_np(posix_spawn_file_actions_t *,
       >         ^
       > 3rd/bee.lua/bee/subprocess/subprocess_posix.cpp:200:63: error: cannot initialize a parameter of type 'int' with an lvalue of type 'const char *'
       >             posix_spawn_file_actions_addchdir_np(&spawnfile_, cwd);
       >                                                               ^~~
       > /nix/store/x9qw47hy8mpczwbvb03vyyn8qb2v27n7-Libsystem-1238.60.2/include/spawn.h:144:6: note: passing argument to parameter here
       >                 int) __OSX_AVAILABLE_STARTING(__MAC_10_7, __IPHONE_4_3);
       >                    ^
       > 2 errors generated.
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/lzp8567dcngr7g69h8468d858aihs15j-sumneko-lua-language-server-3.1.0.drv'.
```

### Notify maintainers
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@sei40kr the last version bump https://github.com/NixOS/nixpkgs/pull/168929 broke darwin builds

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-darwin"`
 - host os: `Darwin 21.4.0, macOS 10.16`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.7.0`
 - channels(lockejan): `"home-manager, nixpkgs"`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/Users/lockejan/.nix-defexpr/channels/nixpkgs`

```

3.0.2 has the same build problem:

```
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/b5lza9njxn7526r7dlrb7zgrmdnbk577-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
building
build flags: -j12 -l12 -fcompile/ninja/macos.ninja
[1/60] Compile C++ build/macos/obj/source_bee/simplethread_posix.obj
[2/60] Compile C   build/macos/obj/source_bee/lua-seri.obj
[3/60] Compile C++ build/macos/obj/source_bee/error.obj
[4/60] Compile C++ build/macos/obj/source_bee/socket.obj
[5/60] Compile C++ build/macos/obj/source_bee/os.obj
[6/60] Compile C++ build/macos/obj/source_bee/file_handle.obj
[7/60] Compile C++ build/macos/obj/source_bee/file_handle_posix.obj
[8/60] Compile C++ build/macos/obj/source_bee/file_handle_osx.obj
[9/60] Compile C++ build/macos/obj/source_bee/endpoint.obj
[10/60] Compile C++ build/macos/obj/source_bee/path_helper.obj
[11/60] Compile C++ build/macos/obj/source_bee/subprocess_posix.obj
FAILED: build/macos/obj/source_bee/subprocess_posix.obj 
clang++ -MMD -MT build/macos/obj/source_bee/subprocess_posix.obj -MF build/macos/obj/source_bee/subprocess_posix.obj.d -std=c++17 -O2 -Wall -Werror -fvisibility=hidden -mmacosx-version-min=10.15 -I3rd/bee.lua/bee/nonstd -I3rd/bee.lua -DBEE_INLINE -DNDEBUG -Wunguarded-availability -o build/macos/obj/source_bee/subprocess_posix.obj -c 3rd/bee.lua/bee/subprocess/subprocess_posix.cpp
3rd/bee.lua/bee/subprocess/subprocess_posix.cpp:200:13: error: use of undeclared identifier 'posix_spawn_file_actions_addchdir_np'; did you mean 'posix_spawn_file_actions_addinherit_np'?
            posix_spawn_file_actions_addchdir_np(&spawnfile_, cwd);
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            posix_spawn_file_actions_addinherit_np
/nix/store/x9qw47hy8mpczwbvb03vyyn8qb2v27n7-Libsystem-1238.60.2/include/spawn.h:143:5: note: 'posix_spawn_file_actions_addinherit_np' declared here
int     posix_spawn_file_actions_addinherit_np(posix_spawn_file_actions_t *,
        ^
3rd/bee.lua/bee/subprocess/subprocess_posix.cpp:200:63: error: cannot initialize a parameter of type 'int' with an lvalue of type 'const char *'
            posix_spawn_file_actions_addchdir_np(&spawnfile_, cwd);
                                                              ^~~
/nix/store/x9qw47hy8mpczwbvb03vyyn8qb2v27n7-Libsystem-1238.60.2/include/spawn.h:144:6: note: passing argument to parameter here
                int) __OSX_AVAILABLE_STARTING(__MAC_10_7, __IPHONE_4_3);
                   ^
2 errors generated.
[12/60] Compile C++ build/macos/obj/source_bee/lua_platform.obj
[13/60] Compile C++ build/macos/obj/source_bee/lua_serialization.obj
[14/60] Compile C++ build/macos/obj/source_bee/lua_socket.obj
[15/60] Compile C++ build/macos/obj/source_bee/lua_time.obj
[16/60] Compile C++ build/macos/obj/source_bee/path_helper-1.obj
[17/60] Compile C++ build/macos/obj/source_bee/lua_filesystem.obj
[18/60] Compile C++ build/macos/obj/source_bee/lua_filewatch.obj
[19/60] Compile C++ build/macos/obj/source_bee/lua_subprocess.obj
[20/60] Compile C++ build/macos/obj/source_bee/fsevent_osx.obj
[21/60] Compile C++ build/macos/obj/source_bee/lua_thread.obj
[22/60] Compile C++ build/macos/obj/source_bee/format.obj
ninja: build stopped: subcommand failed.

```

--- a/pkgs/development/tools/sumneko-lua-language-server/default.nix
+++ b/pkgs/development/tools/sumneko-lua-language-server/default.nix
@@ -24,6 +24,13 @@ stdenv.mkDerivation rec {
     darwin.apple_sdk.frameworks.Foundation
   ];
 
+  # Disable cwd support on darwin, because it requires macOS>=10.15
+  preConfigure = lib.optionalString stdenv.isDarwin ''
+    for file in 3rd/bee.lua/bee/subprocess/subprocess_posix.cpp 3rd/luamake/3rd/bee.lua/bee/subprocess/subprocess_posix.cpp; do
+      substituteInPlace $file --replace '#define SUPPORT_CWD 1' ""
+    done
+  '';
+
   preBuild = ''
     cd 3rd/luamake
   ''

